main.go

```diff
@@ -0,0 +1,205 @@
+package main
+
+import (
+	"code.cloudfoundry.org/cli/plugin"
+	"context"
+	"flag"
+	"fmt"
+	"github.com/cloudfoundry/go-cfclient/v3/client"
+	"github.com/cloudfoundry/go-cfclient/v3/config"
+	"github.com/cloudfoundry/go-cfclient/v3/resource"
+	"log"
+	"net/url"
+	"os"
+	"strings"
+)
+
+type lookupRoute struct{}
+
+func main() {
+	plugin.Start(new(lookupRoute))
+}
+
+func (l lookupRoute) Run(cliConnection plugin.CliConnection, args []string) {
+	var err error
+	defer func() {
+		if err != nil {
+			fmt.Printf("error: %s\n", err.Error())
+			os.Exit(1)
+		}
+	}()
+
+	log.SetFlags(0)
+	flags := flag.NewFlagSet("lookup-route", flag.ContinueOnError)
+	target := flags.Bool("t", false, "Target the org/space containing this route")
+	err = flags.Parse(args[1:])
+	if err != nil {
+		return
+	}
+
+	if len(flags.Args()) == 0 || len(args) == 0 {
+		err = fmt.Errorf("please specify the required parameters")
+		return
+	}
+	hostName := flags.Args()[0]
+
+	hasApiEndpoint, err := cliConnection.HasAPIEndpoint()
+	if err != nil || !hasApiEndpoint {
+		err = fmt.Errorf("no API endpoint set")
+		return
+	}
+
+	cfc, err := createCfClient()
+	if err != nil {
+		return
+	}
+
+	route, err := findRoute(cfc, hostName)
+	if err != nil {
+		return
+	}
+
+	err = lookup(cfc, route, *target, cliConnection)
+	if err != nil {
+		return
+	}
+}
+
+func (l lookupRoute) GetMetadata() plugin.PluginMetadata {
+	return plugin.PluginMetadata{
+		Name: "lookup-route",
+		Version: plugin.VersionType{
+			Major: 0,
+			Minor: 1,
+			Build: 0,
+		},
+		Commands: []plugin.Command{
+			{
+				Name:     "lookup-route",
+				HelpText: "Cloud Foundry CLI plugin to identify the application, a given route is pointing to.",
+				UsageDetails: plugin.Usage{
+					Usage: "cf lookup-route [-t] ROUTE_URL",
+					Options: map[string]string{
+						"t": "Target the org/space containing the route",
+					},
+				},
+			},
+		},
+	}
+}
+
+func createCfClient() (*client.Client, error) {
+	cfg, err := config.NewFromCFHome()
+	if err != nil {
+		return &client.Client{}, err
+	}
+
+	cfc, err := client.New(cfg)
+	if err != nil {
+		return &client.Client{}, err
+	}
+
+	return cfc, nil
+}
+
+func retrieveDomains(cfc *client.Client, domainName string) ([]*resource.Domain, error) {
+	domainOpts := client.NewDomainListOptions()
+	domainOpts.Names.Values = append(domainOpts.Names.Values, domainName)
+	domains, err := cfc.Domains.ListAll(context.Background(), domainOpts)
+	if err != nil {
+		return nil, err
+	}
+	return domains, nil
+}
+
+func findDomain(cfc *client.Client, query string) (*resource.Domain, string, *url.URL, error) {
```

Comment:
The function name implies that this only looks up the domain but it returns the domain a string and a URL. It seems like we should maybe require the query to be passed in as a URL already if you need that in multiple places.

Reply:
renamed the func